### ISSUE TITLE
jsdoc/check-param-names incorrectly failing when using default params

### ISSUE THREAD

**Issue by OmgImAlexis** [gajus/eslint-plugin-jsdoc#377]
The error I'm getting.
```
  ⚠  632:0  Expected @param names to be "name". Got "name ".                                     jsdoc/check-param-names
```

```js
	/**
     * Loads server with { async start(), async stop() } signature
     *
     * @name core.loadServer
     * @param {string} [name = 'server'] The name of the unraid server to load.
     */
	const loadServer = async(name = 'server') => {};
```

**Comment by gajus** [comment]
```js
/**
 * Loads server with { async start(), async stop() } signature
 *
 * @name core.loadServer
 * @param {string} [name='server'] The name of the unraid server to load.
 */

```

^would work

**Comment by OmgImAlexis** [comment]
I have another rule that clashes as it requires spaces like this `[node = 'server]`.

**Comment by gajus** [comment]
Sounds like a good first contribution.

**Comment by OmgImAlexis** [comment]
@gajus would this be the "correct" way of fixing this? If so I'll open a PR.

```diff
diff --git a/src/rules/checkParamNames.js b/src/rules/checkParamNames.js
index 72cf61f..6675e35 100644
--- a/src/rules/checkParamNames.js
+++ b/src/rules/checkParamNames.js
@@ -40,7 +40,7 @@ const validateParameterNames = (targetTagName : string, functionParameterNames :
       return false;
     }
 
-    if (functionParameterName !== tag.name) {
+    if (functionParameterName !== tag.name.trim()) {
       const expectedNames = functionParameterNames.join(', ');
       const actualNames = paramTags.map(([, {name}]) => {
         return name;
```

Before:
```
  utils/vms/parse-domain/sanitize-vendor.js:11:NaN
  ⚠  11:0   Expected @param names to be "vendorName". Got "vendorName ".                         jsdoc/check-param-names

  utils/misc/exit-app.js:15:21
  ⚠  13:0   Expected @param names to be "error, code". Got "error, code ".                       jsdoc/check-param-names
```

After:
No errors.

**Comment by brettz9** [comment]
Looks great to me. If you could just add the passing test, along with this config (assuming you are using `async` in the example), it should be good to go:

```js
      parserOptions: {
        ecmaVersion: 2017,
      },
```

(While we could look at getting `comment-parser` to support the trimming of names itself, I think we'd probably rather not have it trim them for the sake of our fixers being able to leave such spacing alone.)
